### PR TITLE
Integrate DetailviewExtension into multi-select views

### LINKED ISSUE #3072: Enhance DetailViewExtension hook to multi-select views
Currently there is no good way to add custom content or options to a multi-select view in monitoring.

I would add this as extra function to the DetailViewExtension.

```php
public function getHtmlForObjects($objects) {}
```

Notes:
* Can't be abstract so its compatible
* Multi select views need to be extended for hooks

### PR DESCRIPTION
**Pull request by lazyfrosch**
Also make sure Javascript module handling will find the container for module auto-loading.

@lippserd Please review if this makes sense.

fixes #3072

### CODE REVIEW

**Comment by lazyfrosch** [comment]
Here is an example that utilizes these updated hook:

bildschirmfoto von 2018-01-23 17-31-35
bildschirmfoto von 2018-01-23 17-31-26

**Comment by lazyfrosch** [comment]
@lippserd please review and merge

File: modules/monitoring/application/controllers/HostsController.php [hunk 1]
```diff
@@ -151,6 +153,16 @@ public function showAction()
         $this->view->commentsLink = Url::fromRequest()->setPath('monitoring/list/comments');
         $this->view->sendCustomNotificationLink = Url::fromRequest()
             ->setPath('monitoring/hosts/send-custom-notification');
+
+        $this->view->extensionsHtml = array();
+        foreach (Hook::all('Monitoring\DetailviewExtension') as $hook) {
+            /** @var DetailviewExtensionHook $hook */
+            $module = $this->view->escape($hook->getModule()->getName());
+            $this->view->extensionsHtml[] =
```
Comment (lippserd) [inline modules/monitoring/application/controllers/HostsController.php]: The extension should only be added to the array if the hook does not return the empty string. Else we may render empty containers.

File: modules/monitoring/application/controllers/ServicesController.php [hunk 2]
```diff
@@ -153,6 +155,16 @@ public function showAction()
         $this->view->sendCustomNotificationLink = Url::fromRequest()->setPath(
             'monitoring/services/send-custom-notification'
         );
+
+        $this->view->extensionsHtml = array();
+        foreach (Hook::all('Monitoring\DetailviewExtension') as $hook) {
+            /** @var DetailviewExtensionHook $hook */
+            $module = $this->view->escape($hook->getModule()->getName());
+            $this->view->extensionsHtml[] =
```
Comment (lippserd) [inline modules/monitoring/application/controllers/ServicesController.php]: The extension should only be added to the array if the hook does not return the empty string. Else we may render empty containers.

**Comment by lazyfrosch** [comment]
Rebased and updated to your suggestions.

I also copied the "only output if content" check to the normal DetailviewExtension usage.

**Comment by lippserd** [comment]
@nilmerg Your feedback is still pending here.

**Pull request merged**